Framework/PythonInterface/mantid/FunctionWrapper.py

```diff
@@ -0,0 +1,480 @@
+      
+          :param name: name that appears in the []
+          :param value: new value of this item
+      """
+      if type(name) == type('string') and self.fun.hasAttribute(name):
+         self.fun.setAttributeValue(name, value)
+      else:
+         self.fun.setParameter(name, value)
+      
+  def __str__ (self):
+      """ Return string giving contents of function.
+          Used in unit tests.
+      """
+      return self.fun.__str__()
+                 
+  def __add__ (self, other):
+      """ Implement + operator for composite function
+      
+          :param other: functionWrapper to be added to self
+      """
+      sum = CompositeFunctionWrapper(self, other)
+      if sum.pureAddition:
+         sum = sum.flatten()
+      return sum     
+        
+  def __mul__ (self, other):
+      """ Implement * operator for product function
+      
+          :param other: functionWrapper to multiply self by
+      """
+      prod = ProductFunctionWrapper(self, other)
+      if prod.pureMultiplication:
+         prod = prod.flatten()
+      return prod
+        
+  def tie (self, *args, **kwargs):
+    """ Add ties.
+    """
+    for a in args:
+        if isinstance(a, dict):
+          for key in a:
+            self.fun.tie(key, str(a[key]))
+
+    for key in kwargs:
+        self.fun.tie(key, str(kwargs[key]))
+        
+  def fix(self, name):
+      """ Fix a parameter.
+      
+          :param name: name of parameter to be fixed
+      """
+      self.fun.fixParameter(name)
+      
+  def fixAllParameters(self):
+       """ Fix all parameters.
+       """
+       for i in range(0, self.fun.numParams()):
+          self.fix(self.getParameterName(i))
+      
+  def untie(self, name):
+      """ Remove tie from parameter.
+      
+          :param name: name of parameter to be untied
+      """
+      self.fun.removeTie(name)
+      
+  def untieAllParameters(self):
+      """ Remove ties from all parameters.
+      """
+      for i in range(0, self.fun.numParams()):  
+          self.fun.removeTie(self.getParameterName(i))  
+
+          
+  def constrain(self, expressions):
+      """ Add constraints
+      
+          :param expressions: string of tie expressions      
+      """
+      self.fun.addConstraints( expressions )
+      
+  def unconstrain(self, name):
+      """ Remove constraints from a parameter
+      
+          :param name: name of parameter to be unconstrained
+      """
+      self.fun.removeConstraint(name)
+           
+  def free(self, name):
+      """ Free a parameter from tie or constraint
+      
+          :param name: name of parameter to be freed
+      """
+      self.fun.removeTie(name)
+      self.fun.removeConstraint(name)
+      
+  def getParameterName(self, index):
+      """ Get the name of the parameter of given index
+      
+          :param index: index of parameter
+      """
+      return self.fun.getParamName(index)
+      
+  def getFunction(self):
```

Comment:
This method does not  make sense because one can access the underlying function by accessing attribute `fun` directly from the `FunctionWrapper` instance. If you intent is to protect attribute `fun` from user tampering, then I suggest to change attribute `fun` to `_fun` to mark it as protected and then write only a getter property:
```
@property
def function(self):
    return self._fun
```
The code below shows that assignment to FunctionWrapper.function is unsuccessful:
```
g = msapi.FunctionWrapper( "Gaussian", Height=7.5, Sigma=1.2, PeakCentre=10)
print(g.function)
g2 = msapi.FunctionWrapper( "Gaussian", Height=8.5, Sigma=1.2, PeakCentre=11)
g.function = g2.fun  # unsuccessful assignment 
print(g.function)
g.fun = g2.fun  # successful assignment
print(g.function)
```
Results in:
```
name=Gaussian,Height=7.5,PeakCentre=10,Sigma=1.2
name=Gaussian,Height=7.5,PeakCentre=10,Sigma=1.2
name=Gaussian,Height=8.5,PeakCentre=11,Sigma=1.2
```